ddtrace/contrib/trace_utils.py

```diff
@@ -69,43 +69,72 @@ def int_service(pin, config, default=None):
     # Config is next since it is also configured via code
     # Note that both service and service_name are used by
     # integrations.
-    if "service" in config and config.service is not None:
-        return config.service
-    if "service_name" in config and config.service_name is not None:
-        return config.service_name
+    if "service" in int_config and int_config.service is not None:
+        return int_config.service
+    if "service_name" in int_config and int_config.service_name is not None:
+        return int_config.service_name
 
-    global_service = config.global_config._get_service()
+    global_service = int_config.global_config._get_service()
     if global_service:
         return global_service
 
-    if "_default_service" in config and config._default_service is not None:
-        return config._default_service
+    if "_default_service" in int_config and int_config._default_service is not None:
+        return int_config._default_service
 
     return default
 
 
-def ext_service(pin, config, default=None):
+def ext_service(pin, int_config, default=None):
     """Returns the service name for an integration which is external
     to the application. External meaning that the integration generates
     spans wrapping code that is outside the scope of the user's application. Eg. A database, RPC, cache, etc.
     """
-    config = config or {}
+    int_config = int_config or {}
 
     if pin and pin.service:
         return pin.service
 
-    if "service" in config and config.service is not None:
-        return config.service
-    if "service_name" in config and config.service_name is not None:
-        return config.service_name
+    if "service" in int_config and int_config.service is not None:
+        return int_config.service
+    if "service_name" in int_config and int_config.service_name is not None:
+        return int_config.service_name
 
-    if "_default_service" in config and config._default_service is not None:
-        return config._default_service
+    if "_default_service" in int_config and int_config._default_service is not None:
+        return int_config._default_service
 
     # A default is required since it's an external service.
     return default
 
 
+def is_error_code(status_code):
+    """Returns a boolean representing whether or not a status code is an error code.
+    Error status codes by default are 500-599.
+    You may also enable custom error codes::
+
+        from ddtrace import config
+        config.http_server.error_statuses = '401-404,419'
+
+    Ranges and singular error codes are permitted and can be separated using commas.
+    """
+
+    if not isinstance(config.http_server.error_statuses, str):
+        log.debug("Error statuses in config are not a string")
+        return False
+
+    error_str = config.http_server.error_statuses.strip()
+    error_ranges = error_str.split(",")
```

Comment:
We can probably compute this once when the configuration option is set so that we don't have to build it on each invocation. What do you think @Isabella-Pham?

Reply:
Yup I agree :D